include/devicetree/pwms.h

```diff
@@ -101,6 +101,80 @@ extern "C" {
  */
 #define DT_PWMS_LABEL(node_id) DT_PWMS_LABEL_BY_IDX(node_id, 0)
 
+/**
+ * @brief Get a pwm-controller device object from a pwms property at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_DEVICE_BY_IDX(DT_NODELABEL(n), 0) // "PWM_1"
+ *     DT_PWMS_DEVICE_BY_IDX(DT_NODELABEL(n), 1) // "PWM_2"
+ *
+ * @param node_id node identifier for a node with a pwms property
+ * @param idx logical index into pwms property
+ * @return  A pointer to the pwm-controller device object for the node
+ *          referenced at index "idx"
+ * @see DT_PROP_BY_PHANDLE_IDX()
+ */
+#define DT_PWMS_DEVICE_BY_IDX(node_id, idx) \
```

Comment:
Good additions. I think, however, that these should be more aligned to the corresponding `include/devicetree/gpio.h` macros are named and their parameters (e.g. `DT_PWMS_CTLR_BY_IDX(node_id, pwm_pha, idx)`.

Reply:
Agreed.

These also need to be fixed so they return node identifiers, not devices. The device layer is on top of the devicetree layer and we never mix 'device' level stuff into devicetree-only headers as is being done here.

The docstrings all need to say that these return node identifiers, the code needs to match, and the examples need to be fixed accordingly.

Reply:
@henrikbrixandersen, the idea was to get the device, not the node identifier.
It seems that it does not fit this header. However, adding the CTRL macro for now.